### PR TITLE
editoast, front, core: remove gamma type

### PR DESCRIPTION
**Pull request by bougue-pe**
Part of https://github.com/OpenRailAssociation/osrd/issues/9708

Rename `gamma` to `const_gamma` to reflect its use: constant gamma braking coefficient used when NOT circulating under ETCS/ERTMS signaling system in m/s^2.

All rolling stock used are `CONST`, front [always provided `CONST`](https://github.com/OpenRailAssociation/osrd/pull/9887/files#diff-a720b5041d0ecae646392425aae1d15acb49a0b9749fa9fad25161de0226551aL158-L161), this `MAX` variant was never used, and actually always constant in code use (and not sure if a max-check was missing in [computeAcceleration](https://github.com/OpenRailAssociation/osrd/pull/9887/files#diff-ab3c0ecd0ebc92a826fb7aebe46ecfc25ccf08f0ecd7a4a3f33dc363ea1434b5R105))

🔍 please review by commit.

To merge right after this:
- [x] corresponding PR in dags https://github.com/osrd-project/osrd-dags/pull/596
- [x] probably corresponding PR in https://github.com/y-plus/pyOSRD/pull/113 > useless as pinned to a previous version

### CODE REVIEW

**Review by leovalais — APPROVED** [review]

**Review by Maymanaf — APPROVED** [review]

**Review by Wadjetz — APPROVED** [review]

**Review by Erashin — APPROVED** [review]
Core changes work for me :)

**Review by woshilapin — APPROVED** [review]
The PR seems good. However, I'm missing some understanding about why it's fine to remove some (possibly not really well-supported) functionality: why removing `MAX`? It'd be nice to at least explain in the PR description why it was there? Why we decided to remove it? Just put some tracing about this decision so "future us" understand why, at this point in time, we decided to remove `MAX`.

**Review by shenriotpro — APPROVED** [review]
Thanks

**Review by emersion — APPROVED** [review]
LGTM for `front/` changes (not tested)

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java [hunk 1]
```diff
@@ -53,9 +53,6 @@ static double getMaxEffort(double speed, TractiveEffortPoint[] tractiveEffortCur
     /** The maximum constant deceleration, in m/s^2 */
     double getDeceleration();
 
-    /** The maximum braking force which can be applied at a given speed, in newtons */
-    double getMaxBrakingForce(double speed);
```
Comment (bougue-pe) [inline core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java:60]: @woshilapin wrote:
> The PR seems good. However, I'm missing some understanding about why it's fine to remove some (possibly not really well-supported) functionality: why removing `MAX`? It'd be nice to at least explain in the PR description why it was there? Why we decided to remove it? Just put some tracing about this decision so "future us" understand why, at this point in time, we decided to remove `MAX`.

Added this commit to actually clean the code a bit more after your review.
Also added a little word on main comment of the PR.
🙏 Thanks !

**Review by achrafmohye — APPROVED** [review]
LGTM 👍

File: core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java (hunk 1 above)
Comment (woshilapin) [reply core/envelope-sim/src/main/java/fr/sncf/osrd/envelope_sim/PhysicsRollingStock.java:60]: Thank you 😄

**Review by Erashin — APPROVED** [review]

**Pull request merged**